src/argilla/client/feedback/dataset/local/dataset.py

```diff
@@ -247,6 +259,26 @@ def add_metadata_property(
         self._metadata_properties.append(metadata_property)
         return metadata_property
 
+    def vector_settings_by_name(self, name: str) -> VectorSettings:
+        vector_settings = self._vector_settings.get(name)
+        if not vector_settings:
+            raise KeyError(f"Vector settings with name '{name!r}' does not exist in the dataset.")
+
+        return vector_settings
```

Comment:
Not sure if raising an exception here is too radical. Maybe we should return `None` instead?

```suggestion
        return self._vector_settings.get(name)
```

Reply:
This is aligned to other `*_by_name` methods that we have but could make sense to change the signature to simplify conditional creation workflows (if this setting is not created, then create it, instead of try-catch and create on catch block)

I've created an issue for this. See https://github.com/argilla-io/argilla/issues/4101